src/main/java/core/basesyntax/service/impl/ReaderServiceImpl.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.ReaderService;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+
+public class ReaderServiceImpl implements ReaderService {
+    @Override
+    public List<String> readFile(String filePath) {
+        if (filePath == null || !filePath.endsWith("csv")) {
+            throw new RuntimeException("Incorrect data");
+        }
+        List<String> list;
+        try {
+            list = Files.readAllLines(Path.of(filePath));
+        } catch (IOException ex) {
+            throw new RuntimeException();
```

Comment:
Consider adding some descriptive message to the exception. It is a good practice to include the original exception's message and stack trace when rethrowing it. This helps in debugging.